test/dotnet/Integration/IntegrationTestHelpers.cs

```diff
@@ -82,6 +85,64 @@ void functionLoadedHandler(object sender, DataReceivedEventArgs e)
             return functionsProcess;
         }
 
+        internal static Process StartRedis(string versionPath)
+        {
+            ProcessStartInfo info = new ProcessStartInfo
+            {
+                FileName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? @"C:\Windows\System32\wsl.exe" : $"{versionPath}/src/redis-server",
+                Arguments = $"{(RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? $"{versionPath}/src/redis-server " : "")}--port 6379 --notify-keyspace-events AKE",
+                WindowStyle = ProcessWindowStyle.Hidden,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false
+            };
+
+            Process redisProcess = new Process() { StartInfo = info };
+
+            TaskCompletionSource<bool> hostStarted = new TaskCompletionSource<bool>();
+            void hostStartupHandler(object sender, DataReceivedEventArgs e)
+            {
+                if (e.Data?.Contains("* Ready to accept connections") ?? false)
+                {
+                    hostStarted.SetResult(true);
+                }
+            }
+            redisProcess.OutputDataReceived += hostStartupHandler;
+
+            redisProcess.Start();
+            redisProcess.BeginOutputReadLine();
+            redisProcess.BeginErrorReadLine();
+            if (!hostStarted.Task.Wait(TimeSpan.FromMinutes(1)))
+            {
+                StopRedis(redisProcess);
+                throw new Exception("Redis did not start");
```

Comment:
If there are anything from stderr, consider to mention here. This will help quickly find out what went wrong